### PR TITLE
SAM-2735: Event log showing events for removed assessments

### PR DESCRIPTION
**Pull request by ramonchu2k**

### CODE REVIEW

File: samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/bean/author/EventLogBean.java [hunk 1]
```diff
@@ -126,5 +128,14 @@ public void setFilteredUser(String filteredUser) {
...
       this.assessments = assessments;
    }
+   public Map<Long, Integer> getStatusMap() {
+	   return statusMap;
+   }
+   public void setStatusMap(Map<Long, Integer> statusMap) {
+	   this.statusMap = statusMap;
+   }
+   public boolean isDeleted(Long id){
+	   return statusMap.get(id)==2;
```
Comment (bjones86) [inline samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/bean/author/EventLogBean.java]: What is `2` here? Should probably be a constant with a name clarifying it's role/purpose/etc.

File: samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/listener/author/EventLogListener.java [hunk 2]
```diff
@@ -91,6 +91,18 @@ public void processValueChange(ValueChangeEvent event)
 		return updatedEventLogDataList;
 	}
 
+	private Map<Long,Integer> setStatusEventLog(EventLogService eventLogService, String siteId, EventLogBean eventLog) {
+		List<EventLogData> eventLogDataList = eventLogService.getEventLogData(siteId, -1L, "");
+		PublishedAssessmentService assessmentService = new PublishedAssessmentService();	
+		Map<Long,Integer> statusMap = new  HashMap<Long,Integer>();
```
Comment (bjones86) [inline samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/listener/author/EventLogListener.java]: Diamond operator

File: samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/listener/author/EventLogListener.java [hunk 3]
```diff
@@ -145,14 +160,14 @@ private void processPageLoad(EventLogBean eventLog) {
...
-	private List<SelectItem> initAssessmentListFilter(List<Object[]> assessmentTitles) {
+	private List<SelectItem> initAssessmentListFilter(List<Object[]> assessmentTitles, Map<Long,Integer> statusMap) {
 	   List<SelectItem> assessments = new ArrayList<SelectItem>();
 	   
 	   String allStr = ContextUtil.getLocalizedString("org.sakaiproject.tool.assessment.bundle.EventLogMessages", "filterAll");
       assessments.add(new SelectItem((long)-1, allStr));
       for (Object[] assessment : assessmentTitles) {
          Long id = (Long)assessment[0];
-         String title = (String)assessment[1];
+         String title = (String)assessment[1]+((statusMap.get(id)!=null && statusMap.get(id)==2)?"-deleted":"");
```
Comment (bjones86) [inline samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/listener/author/EventLogListener.java]: You're performing the `.get( id )` on the `HashMap<>` twice here; this is inefficient. Instead, extract the return value and check the value:

```
Integer status = statusMap.get( id );
String strStatus = status != null && status == 2 ? "-deleted" : "";
String title = (String) assessment[1] + strStatus;
```

File: samigo/samigo-app/src/webapp/jsf/event/eventLog.jsp [hunk 4]
```diff
@@ -77,7 +77,7 @@
 	  </f:facet>
 
 	 <h:panelGroup>
-	  <h:outputText value="#{log.shortenedTitle}"/><h:outputText rendered="#{log.title != log.shortenedTitle}" value="#{eventLogMessages.dotdotdot}"/>
+	  <h:outputText value="#{log.shortenedTitle}"/><h:outputText rendered="#{log.title != log.shortenedTitle}" value="#{eventLogMessages.dotdotdot}"/><h:outputText value="-deleted" rendered="#{eventLog.isDeleted(log.assessmentId)}"/>
```
Comment (bjones86) [inline samigo/samigo-app/src/webapp/jsf/event/eventLog.jsp]: Can you put this new `<h:outputText>` on a new line?

**Comment by ramonchu2k** [comment]
I fixed every suggestion excepting diamond operator. Could you tell me what do you mean with this?
Thanks.

**Comment by bjones86** [comment]
Thanks @ramonchu2k. The diamond operator is new with Java7, and creates less work for the programmer. Originally, you would have to initialize a collection with the same generics that it was defined as:

`Map<Long,Integer> statusMap = new  HashMap<Long,Integer>();`

Now with the diamond operator, the compiler can determine the generics on instantiation by itself:

`Map<Long,Integer> statusMap = new  HashMap<>();`

**Comment by ramonchu2k** [comment]
Thanks for the explanation @bjones86, fixed!

**Pull request merged**